### PR TITLE
Add bin_img to binarize images

### PR DESCRIPTION
**Pull request by dangom**
If there is interest, this PR adds a function to the image package to binarize images. I see myself using it all the time, and thought I'd share it.
It is based off of threshold img and math img to reduce burden on maintainers. As long as those two functions work properly, bin img should as well.

### CODE REVIEW

File: nilearn/image/__init__.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 """
 from .resampling import resample_img, resample_to_img, reorder_img, \
     coord_transform
-from .image import high_variance_confounds, smooth_img, crop_img, \
+from .image import high_variance_confounds, smooth_img, crop_img, bin_img, \
```
Comment (bthirion) [inline nilearn/image/__init__.py:7]:
```suggestion
from .image import high_variance_confounds, smooth_img, crop_img, binarize_img, \
```

Comment (bthirion) [reply nilearn/image/__init__.py:7]: I think that there is little cost in using the full name

File: nilearn/image/image.py [hunk 2]
```diff
@@ -932,6 +932,59 @@ def math_img(formula, **imgs):
...
+        within the range of "0%" to "100%".
+
+    mask_img : Niimg-like object, default None, optional
+        Mask image applied to mask the input data.
+        If None, no masking will be applied.
+
+    Returns
+    -------
+    :class:`~nibabel.nifti1.Nifti1Image`
+        Binarized version of the given input image.
```
Comment (bthirion) [inline nilearn/image/image.py:963]: maybe indicate that the return type is int.

**Review by bthirion — COMMENTED** [review]
Thx. a few comments fore readability.

File: nilearn/image/__init__.py (hunk 1 above)
Comment (dangom) [reply nilearn/image/__init__.py]: fixed

File: nilearn/image/image.py (hunk 2 above)
Comment (dangom) [reply nilearn/image/image.py]: fixed

**Review by NicolasGensollen — COMMENTED** [review]
Thanks @dangom !
Looks good to me. I think we still need to:

- Add `binarize_img` to `doc/modules/reference.rst`
- Add a whatsnew entry
- Build the full doc by including "[circle full]" in the commit message

**Comment by dangom** [comment]
done

**Review by bthirion — APPROVED** [review]
Thx.

**Review by NicolasGensollen — APPROVED** [review]
LGTM, thanks!

**Comment by NicolasGensollen** [comment]
Merging, thanks!

**Pull request merged**